christin/index.js

```diff
@@ -0,0 +1,179 @@
+//*1. Person Class
+//*Create a class called Person which accepts the name of a person as a string, and his / her age as a number.The Person class should have a method called describe which returns a string with the following syntax: "name, age years old".So, for example, if John is 19 years old, then the function describe of his object will return "John, 19 years old".
+
+class Person {
+  constructor(name, age) {
+    this.name = name;
+    this.age = age;
+  }
+
+  describe() {
+    return `${this.name}, ${this.age} years old`;
+  }
+}
+
+const person = new Person("John", 19);
+console.log(person.describe());
+
+// -----------------------------------------------
+//-------------------------###--------------------------
+
+//* 2. Volume
+//* Write a JavaScript program to get the volume of a cylinder with four decimal places using a class. Note: Volume of a cylinder: V = πr ^ 2h - r is the radius and h is the height of the cylinder.
+
+class Volume {
+  constructor(radius, height) {
+    this.radius = radius;
+    this.height = height;
+  }
+
+  getVolume() {
+    let volume = Math.PI * this.radius ** 2 * this.height;
+    return parseFloat(volume).toFixed(4);
+  }
+}
+
+let volume1 = new Volume(3, 6);
+console.log(volume1.getVolume());
+//----------------------------------------###------------------
+
+//*Rewrite the following code to use the "class" syntax.
+
+class Clock {
+  constructor({ template }) {
+    let timer;
+    this.template = template;
+  }
+
+  render() {
+    let date = new Date();
+    let hours = date.getHours();
+    if (hours < 10) hours = "0" + hours;
+
+    let mins = date.getMinutes();
+    if (mins < 10) mins = "0" + mins;
+
+    let secs = date.getSeconds();
+    if (secs < 10) secs = "0" + secs;
+
+    let output = this.template
+      .replace("h", hours)
+      .replace("m", mins)
+      .replace("s", secs);
+
+    console.log(output);
+    return output;
+  }
+
+  stop() {
+    this.stop = function () {
+      clearInterval(timer);
+    };
+    return this.stop;
+  }
+
+  start() {
+    this.start = function () {
+      render = () => {
+        timer = setInterval(render, 1000);
+      };
+    };
+    return this.start;
+  }
+}
+
+let clock = new Clock({ template: "h:m:s" });
+clock.start();
+
```

Comment:
Task 3 ==> Now timer works, if you wanna stop the timer after some time you can add setTimeout method after clock.start()
setTimeout(() => clock.stop(), 4000);   <== in this case we stop the timer after 4seconds